guides/user/canary/judge/index.md

```diff
@@ -0,0 +1,80 @@
+---
+layout: single
+title:  "The Canary Judge&mdash;how does it work?"
+sidebar:
+  nav: guides
+---
+
+{% include toc %}
+
+
+To assess the quality of a canary deployment against a baseline, metrics from
+both deployments are compared in order to check for significant degradation.
+This is done in two phases:
+
+* metric collection (retrieval)
+
+  This phase retrieves the key metrics from the baseline and canary deployments.
+  These metrics are typically stored in a time-series database, and include a
+  set of tags or annotations that identify which deployment the data was
+  collected from (canary or baseline).
+
+* judgment
+
+  In this phase Spinnaker compares those metrics, rendering a decision to pass
+  or fail the canary (that is, was there was a significant degradation in the
+  metrics?) This judgment consists of four main steps, and they're described
+  below.
+
+
+## Step 1: Data validation
+
+Data validation ensures that there's data for the baseline and canary metrics
+before analysis begins.
+
+If metric collection returns an empty array for either the baseline or canary
+metric (or both) the metric is labeled `NODATA` and analysis moves on to the
+next metric.
+
+Note that some metrics might have no data, but for good reasons. For example, an
+error counter has no value if there are no failures. For this reason the judge
+does not automatically fail the canary for `NODATA`.
+
+## Step 2: Data cleaning
+
+This step prepares the raw metrics for comparison. This entails handling missing
+values from the input.
+
+Note that there are different strategies for handling missing values based on
+the type of metric. For example, missing values, represented as NaNs, may be
+replaced with zeros for error metrics while they may be imputed or removed for
+other types of metrics. In addition, the data cleaning step can, optionally,
+remove outliers from the data before comparison.
+
+## Step 3: Metric comparison (classification)
+
+This is the step that compares the canary and baseline data for each included
+metric. The output of this step is a classification, for each metric indicating
+if there is a significant difference between the canary and baseline.
+
+Each metric is classified as either “Pass”, “High”, or “Low” as shown in the
+screen shot below.
+
+<!--- insert image here, if available --->
+
+The primary metric comparison algorithm (classifier) in Kayenta uses a
+nonparametric statistical test to check for a significant difference between the
+canary and baseline metrics.
+
+## Step 4: Score computation
+
+After each metric is classified, a final score is computed which represents how
```

Comment:
I think this isn't quite right and we should double-check this with Chris and Michael. My understanding is that what you describe in this paragraph happens for each metric group (that is, a score is calculated as the ratio of pass metrics over total metrics _in the given group_).

Then a weighted sum is calculated over all the metric group scores, and the resulting number is the final score. That final score is then compare to the thresholds.

Reply:
@csanden 
@skandragon 
What say ye?